app/Dockerfile

```diff
@@ -33,5 +64,76 @@ ENV HOST=0.0.0.0
 # Install application dependencies.
 # https://python-poetry.org/docs/basic-usage/#installing-dependencies
 RUN poetry install
+
 # Run the application.
 CMD ["poetry", "run", "python", "-m", "api"]
+
+#-------------------------------------------------------------------------------
+# Non-development build environment
+#-------------------------------------------------------------------------------
+FROM build as app-build
+
+# Runtime arguement for what user will the bulk of this stage run as
+ARG RUN_USER
+
+# In between ARG RUN_USER and USER ${RUN_USER}, the user is still root
+# If there is anything that needs to be ran as root, this is the spot
+
+# Change the user from root to ${RUN_USER}
+USER ${RUN_USER}
+
+# Set the application working directory.
+WORKDIR /app
+
+# install dependencies
+# use a project-local virtualenv to be easy to find
+ENV POETRY_VIRTUALENVS_IN_PROJECT true
+COPY pyproject.toml poetry.lock ./
+RUN poetry install --no-root --with app --without dev
+
+COPY . ./
```

Comment:
in this stage we do

```
WORKDIR /app
...
COPY . ./
```
but in dev we do
```
WORKDOR /app
COPY . /app
```
could we be consistent between the two (ether `COPY . .` or `COPY . /app`  since they do the same thing? i don't really have a preference between the two, but if we decide to do `COPY . /app` instead of `COPY . .` then let's also change the `COPY pyproject.toml poetry.lock ./` line to be `COPY pyproject.toml poetry.lock /app`)

Reply:
I'll run the bottom version to see if there was a reason, and then make a commit with whichever version works in both cases, I forget why I did it like that